### ISSUE TITLE
[BUG] K3S New-AksEdgeDeployment fails on current PowerShell versions (7.3.x) due to incorrect command line spacing 

### ISSUE THREAD

**Issue by JuergenSchwertl** [Azure/AKS-Edge#34]
**Describe the bug**
With PoweShell 7.x (most) calls to nodectl.exe fail with error "unknown command "" ..."
_C:\Program Files\AksEdge\nodectl.exe  security login --loginpath c:\programdata\wssdagent\nodelogin.yaml --identity failed to execute [Error: **unknown command "" for "nodectl security login**"]
Initialization of host services failed: False OK at line 2608_

**To Reproduce**

- Start Windows Powershell and follow instructions to "Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows"
- Install PowerShell 7.3.1 
- 	https://github.com/PowerShell/PowerShell/releases/download/v7.3.1/PowerShell-7.3.1-win-x64.msi
- Install AKS/K3S as suggested ( https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-howto-setup-machine ): 
- 	**msiexec.exe /i AksEdge-K3s-0.7.22335.1024.msi** 
- Clone github.com/Azure/Aks-Edge
- Configure ./tools./aksedge-config.json as appropriate
- Run **AksEdgePrompt-PS7.cmd** as admin
- **New-AksEdgeDeployment -JsonConfigFilePath .\aksedge-config.json**
- Screen output: see below

**Additional context**
Looking at the module code of "C:\Program Files\WindowsPowerShell\Modules\AksEdge\0.7.22335.1024\AzureEFLOW.psm1" Line 6223:
   **Execute-NodeCommand $(" security login --loginpath "+$script:nodeloginYAMLLocation+" --identity")**
the nodectl parameters are indented by a space and the downlevel execute-Command function then splits it in line 5227
   **$argumentsToUse = $arguments.Split(" ")**
leading to the first array-member being empty. The PowerShell 7 call parser ( & ) apparently directly parses the command line arg vector into nodectl.exe::main() and argv[0] is then NULL leading to nodectl.exe bailing out with empty command error "unknown command "" for ..." 
Searching AzureEFLOW.psm for 'Execute-NodeCommand " ' (**with** leading space character after command line string start) shows 28 occurances that will break due to command line being split on space. This will also break for commandline arguments on paths with space character (i.e. yaml files in standard AKS install directory "c:\Program Files\Aks-Edge\yaml..." ) 

**Screen output**
PowerShell 7.3.1
Loading AksEdgeDeploy module from C:\Repos\AKS-Edge\tools\modules..
Hyper-V is enabled
HostOS  : Microsoft Windows 11 Enterprise(4)
Version : 10.0.22621.963
Lang    : en-US
Name    : js-de-aks-node
Total CPUs              : 4
Free RAM / Total RAM    : 12 GB / 16 GB
Free Disk / Total Disk  : 102 GB / 126 GB
Running as a virtual machine in Azure environment (Name= js-de-aks-nodevmSize= Standard_D4s_v3offer= windows-11sku= win11-22h2-ent )with Nested Hyper-V enabled
AKS Edge Essentials - K3s (Public Preview) 0.7.22335.1024 is installed.
Loading AksEdge module..
AksEdge version         : 0.7.22335.1024
AksEdgeShell  version   : 1.0.221212.1200
AksEdgeDeploy version   : 1.0.221212.1200
PS C:\Repos\AKS-Edge\tools> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.1
PSEdition                      Core
GitCommitId                    7.3.1
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Repos\AKS-Edge\tools> New-AksEdgeDeployment -JsonConfigFilePath .\aksedge-config.json
 - Configuration does not exist. Creating a default configuration...

[01/03/2023 10:52:21] Creating configuration

 - Configuration created successfully

[01/03/2023 10:52:21] *** No errors found in the Deployment Configuration.

[01/03/2023 10:52:23] AksEdge - deploying a new Linux single machine k3s cluster

[01/03/2023 10:52:24] Checking host for required features

 - Checking the status of 'Microsoft-Hyper-V'
 - Checking the status of 'Microsoft-Hyper-V-Management-PowerShell'
 - Checking the status of 'Microsoft-Hyper-V-Hypervisor'
 - Checking the status of 'OpenSSH.Client*'

[01/03/2023 10:52:26] Creating single machine cluster hcs network

 - AksEdge - private network carved:
Name  : WindowsVmIp4Address
Value : 192.168.0.3

Name  : ip4Subnet
Value : 192.168.0.0

Name  : ip4PrefixLength
Value : 24

Name  : ip4GatewayAddress
Value : 192.168.0.1

Name  : ServiceIPRangeStart
Value : 192.168.0.4

Name  : LinuxVmIp4Address
Value : 192.168.0.2

Name  : ServiceIPRangeEnd
Value : 192.168.0.23

 - Determining DNS servers capable of resolving the endpoint 'microsoft.com' from the list of potential servers (10.1.0.1 168.63.129.16)...
 - Success, one or more eligible DNS servers found (168.63.129.16)...

[01/03/2023 10:52:39] Deploying AKS Edge Essentials - K3s (Public Preview)

 - WARNING: This is a Public Preview version of AKS Edge Essentials - K3s (Public Preview), not intended for production use.
 - WARNING: A clean install may be required for production use once the final General Availability (GA) release is available.

[01/03/2023 10:52:39] Step 1: Preparing host for AKS Edge Essentials - K3s (Public Preview)

[01/03/2023 10:52:39] Enabling Microsoft Update. This will allow AKS Edge Essentials - K3s (Public Preview) to receive updates.

 - Microsoft Update is enabled.

[01/03/2023 10:52:40] Checking for virtual switch with name 'aksedgesw-int'

 - The virtual switch 'aksedgesw-int' of type 'Internal' is present

[01/03/2023 10:52:43] Associating wssdagent service with nodectl

 _C:\Program Files\AksEdge\nodectl.exe  security login --loginpath c:\programdata\wssdagent\nodelogin.yaml --identity failed to execute [Error: unknown command "" for "nodectl security login"]
 Initialization of host services failed: False OK at line 2608_

[01/03/2023 10:52:45] Collecting logs from deployment...

[01/03/2023 10:52:45] Collecting 'AKS Edge Essentials - K3s (Public Preview)' configuration

[01/03/2023 10:52:45] Collecting 'AKS Edge Essentials - K3s (Public Preview)' event logs

[01/03/2023 10:52:46] Collecting wssdagent configuration

[01/03/2023 10:52:46] Collecting wssdagent logs

[01/03/2023 10:52:46] Collecting node logs

[01/03/2023 10:52:46] Exception caught!!!

_C:\Program Files\AksEdge\nodectl.exe compute vm list -o tsv --query "[*].name" failed to execute [Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority"] at line 5254_

[01/03/2023 10:52:46] Compressing logs

[01/03/2023 10:52:46] Zip file is located at "C:\ProgramData\AksEdge\logs\aksedgelogs-230103-1052.zip"

C:\ProgramData\AksEdge\logs\aksedgelogs-230103-1052.zip

[01/03/2023 10:52:46] Attempting to remove hcs single machine cluster network

 - Cleaning up single machine cluster HNS network 'aksedgesw-int' ...
OK

**Expected behavior**
Installation should succeed

aksedgelogs attached : 
[aksedgelogs-230103-1052.zip](https://github.com/Azure/AKS-Edge/files/10335869/aksedgelogs-230103-1052.zip)

**Comment by JuergenSchwertl** [comment]
Once you get past #35 and setup AKS/K3S from a plain Windows Powershell session, trying to access K3S using [AKS Edge Commands](https://learn.microsoft.com/en-us/azure/aks/hybrid/reference/aks-edge-ps/) mostly fail with Error
_[Error: unknown command "" for "nodectl ..._

Examples (now from "real" dev device):
```
PowerShell 7.3.1
PS C:\Users\jschwert> cd C:\Repos\Azure\AKS-Edge\tools\
PS C:\Repos\Azure\AKS-Edge\tools> .\aksEdgeShell.ps1
Loading AksEdgeDeploy module from C:\Repos\Azure\AKS-Edge\tools\modules..
Hyper-V is enabled
HostOS  : Microsoft Windows 11 Enterprise(4)
Version : 10.0.22621.963
Lang    : en-US de-DE en-GB
Name    : JS-MS-P1
Total CPUs              : 16
Free RAM / Total RAM    : 43 GB / 64 GB
Free Disk / Total Disk  : 712 GB / 952 GB
AKS Edge Essentials - K3s (Public Preview) 0.7.22335.1024 is installed.
Loading AksEdge module..
AksEdge version         : 0.7.22335.1024
AksEdgeShell  version   : 1.0.221212.1200
AksEdgeDeploy version   : 1.0.221212.1200
Current PowerShell is:

Name                           Value
----                           -----
PSVersion                      7.3.1
PSEdition                      Core
GitCommitId                    7.3.1
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Repos\Azure\AKS-Edge\tools> Get-AksEdgeKubeConfig

Confirm
Are you sure you want to perform this action?
Performing the operation "Retrieve and write the cluster kubeconfig file" on target "C:\Users\jschwert\.kube".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): y

[01/03/2023 16:40:36] Exception caught!!!

 - C:\Program Files\AksEdge\nodectl.exe  compute vm list -o tsv --query "[?tags.keys(@).contains(@,'9f0ea5f3-5769-47e3-b504-2afacd1fef0f')].name" failed to execute [Error: unknown command "" for "nodectl compute virtualmachine list"] at C:\Program Files\WindowsPowerShell\Modules\AksEdge\0.7.22335.1024\AzureEFLOW.psm1:5254

PS C:\Repos\Azure\AKS-Edge\tools> Get-AksEdgeNodeName

[01/03/2023 17:06:32] Exception caught!!!
 - C:\Program Files\AksEdge\nodectl.exe  compute vm list -o tsv --query "[?tags.keys(@).contains(@,'9f0ea5f3-5769-47e3-b504-2afacd1fef0f')].name" failed to execute [Error: unknown command "" for "nodectl compute virtualmachine list"] at line 5254

PS C:\Repos\Azure\AKS-Edge\tools> Get-AksEdgeDeploymentInfo
[01/03/2023 17:07:23] Acquiring 'AKS Edge Essentials - K3s (Public Preview)' deployment information
 - Caught exception C:\Program Files\AksEdge\nodectl.exe  compute vm list -o tsv --query "[?tags.keys(@).contains(@,'9f0ea5f3-5769-47e3-b504-2afacd1fef0f')].name" failed to execute [Error: unknown command "" for "nodectl compute virtualmachine list"], while retrieving ARC deployment information
Exception: C:\Program Files\AksEdge\nodectl.exe  compute vm list -o tsv --query
"[?tags.keys(@).contains(@,'9f0ea5f3-5769-47e3-b504-2afacd1fef0f')].name" failed to execute [Error: unknown command ""
for "nodectl compute virtualmachine list"]

PS C:\Repos\Azure\AKS-Edge\tools> Get-AksEdgeNodeAddr
[01/03/2023 17:08:04] Exception caught!!!
 - C:\Program Files\AksEdge\nodectl.exe  compute vm list -o tsv --query "[?tags.keys(@).contains(@,'9f0ea5f3-5769-47e3-b504-2afacd1fef0f')].name" failed to execute [Error: unknown command "" for "nodectl compute virtualmachine list"] at line 5254

[01/03/2023 17:08:04] Exception caught!!!
 - Virtual machine name could not be retrieved, failed to get IP/MAC address at line 3831
```

**Comment by fcabrera23** [comment]
Hi @JuergenSchwertl,

Thanks for your detailed information. We will fix this issue in our next version.

Thanks,
Francisco

**Comment by fcabrera23** [comment]
Hi @JuergenSchwertl,

This issue was fixed in the [1.0.266.0](https://github.com/Azure/AKS-Edge/releases/tag/1.0.266.0) update.

Thanks,
Francisco
